tests/test_model.py

```diff
@@ -3438,3 +3438,18 @@ def test_delete(add_version_condition: bool) -> None:
         }
         args = req.call_args[0][1]
         assert args == expected
+
+@patch.object(target=Model, attribute="exists", autospec=True)
+@patch.object(target=Model, attribute="_get_connection", autospec=True)
+def test_delete_with_wait(mock__get_connection, mock_exists):
```

Comment:
I find [mocker](https://pytest-mock.readthedocs.io/en/latest/usage.html) to be more readable than `@patch` decorators, since you don't need to match arguments in reverse.

```suggestion
def test_delete_with_wait(mocker):
    mock_exists = mocker.patch.object(target=Model, attribute="exists", autospec=True)
    mock__get_connection = mocker.patch.object(target=Model, attribute="_get_connection", autospec=True)
```

Reply:
TIL